tchannel/container/heap.py

```diff
@@ -0,0 +1,99 @@
+from __future__ import absolute_import
+from __future__ import division
+
+import math
+import six
+
+
+class HeapOperation(object):
+    """HeapOperation defines the interface on how to manipulate the heap.
+    Any extension of heap class should implement all the heap operations in
+    order to have complete heap functions.
+    """
+
+    def lt(self, i, j):
+        """Compare the items in position i and j of the heap.
+
+        :param i: the first item's position of the heap list
+        :param j: the second item's position of the heap list
+        :return true if items[i] < items[j], otherwise false.
+        """
+        raise NotImplementedError()
+
+    def push(self, x):
+        """Push a new item into heap"""
+        raise NotImplementedError()
+
+    def pop(self):
+        """Pop an item from the heap"""
+        raise NotImplementedError()
+
+    def swap(self, i, j):
+        """swap items between position i and j of the heap"""
+        raise NotImplementedError()
+
+    def size(self):
+        """Return length of the heap."""
+        raise NotImplementedError()
+
+
+def init(h):
+    # heapify
+    n = h.size()
+    for i in six.moves.range(int(math.floor(n/2)) - 1, -1, -1):
+        down(h, i, n)
+
+
+def push(h, x):
+    h.push(x)
+    up(h, h.size()-1)
+
+
+def pop(h):
+    n = h.size() - 1
+    h.swap(0, n)
+    down(h, 0, n)
+    return h.pop()
+
+
+def remove(h, i):
+    n = h.size() - 1
+    if n != i:
+        h.swap(i, n)
+        down(h, i, n)
+        up(h, i)
+
+    return h.pop()
+
+
+def fix(h, i):
+    down(h, i, h.size())
```

Comment:
Looks like everything except `up` and `down` is the public interface. Can you document the public interface of this module?